ntp-daemon/src/config/mod.rs

```diff
@@ -201,6 +202,13 @@ pub enum ConfigError {
 
 impl Config {
     async fn from_file(file: impl AsRef<Path>) -> Result<Config, ConfigError> {
+        let meta = std::fs::metadata(&file).unwrap();
+        let perm = meta.permissions();
+        
+        if perm.mode() & (libc::S_IRWXO | libc::S_IRWXG | libc::S_IRWXU) == 0o777 {
```

Comment:
Sorry, the original issue wasn't very worked out on this, but thinking about it a bit more, I think we should primarily warn when the config is world writeable, and this can happen even if the permissions arent 0o777 (a somewhat stupid, but valid example of this on linux is a file with permissions 0117).

Reply:
It is best to include this in this PR

Reply:
Done